add candlestick chart

1010Tom opened the pull request:
Kudos to @da-kami who did the initial work

resolves #128 

![simulator_screenshot_CCEBF768-CB8A-4771-84F3-90B1DEE80C45](https://user-images.githubusercontent.com/88702308/225499060-1e561c95-eb75-4eee-9410-7a8c44d0dd0c.png)

1010Tom commented:
@da-kami : do you mind helping out here please: I do not understand the test failure :/

1010Tom commented:
Oh, I think I understand now, it's because flutter doesn not allow http requests in tests.

da-kami commented on mobile/lib/features/trade/candlestick_chart.dart:
```diff
@@ -1,14 +1,78 @@
+    );
+  }
+
+  Future<void> getNewCandles() async {
```
This function should not be in the `widget` code, but should be move to a `Service` - we do fetch data from an external system here.

It would be great if the service is configurable in the widget - that way it can be mocked in the test (that should answer how to handle this widget in the test :)

da-kami reviewed (commented):
Nice that this works with bitmex out of the box

da-kami commented:
> @da-kami : do you mind helping out here please: I do not understand the test failure :/

see my answer here: https://github.com/get10101/10101/pull/270#discussion_r1141496335

1010Tom commented:
@da-kami : I need help with this please. Keen to pair?

holzeis commented:
> @da-kami : I need help with this please. Keen to pair?

Hey @1010Tom, I hope you don't mind. I refactored your code to align with the pattern used for other components and fixed the tests. https://github.com/get10101/10101/pull/270/commits/c6846f7e350c259e3cf80c3384e45c80b1248fcc

1010Tom commented:
> > @da-kami : I need help with this please. Keen to pair?
> 
> Hey @1010Tom, I hope you don't mind. I refactored your code to align with the pattern used for other components and fixed the tests. [c6846f7](https://github.com/get10101/10101/commit/c6846f7e350c259e3cf80c3384e45c80b1248fcc)

nice. Looks good :)

1010Tom commented:
Cool thanks, looks good.

da-kami commented on mobile/lib/features/trade/trade_screen.dart:
```diff
@@ -96,6 +97,7 @@ class TradeScreen extends StatelessWidget {
 
     OrderChangeNotifier orderChangeNotifier = context.watch<OrderChangeNotifier>();
     PositionChangeNotifier positionChangeNotifier = context.watch<PositionChangeNotifier>();
+    CandlestickChangeNotifier candlestickService = context.watch<CandlestickChangeNotifier>();
```
```suggestion
    CandlestickChangeNotifier candlestickChangeNotifier = context.watch<CandlestickChangeNotifier>();
```

1010Tom commented:
bors r+
